### ISSUE TITLE
Implement validation

### ISSUE THREAD

**Issue by jualat** [jualat/CleanRLHF#79]
implement some method to detect overfitting and use it for regularization
- train / test split
- cross entropy loss
- dropout layers (just if we have overfitting)
